test/integration/pod_state_test.go

```
@@ -0,0 +1,147 @@
+
+			// Create nodes and pods.
+			for _, node := range tt.nodes {
+				if _, err := cs.CoreV1().Nodes().Create(testCtx.Ctx, node, metav1.CreateOptions{}); err != nil {
+					t.Fatalf("failed to create node: %v", err)
+				}
+			}
+
+			// Create existing Pods on node.
+			for _, pod := range tt.pods {
+
+				// Create Nominated Pods by setting two pods exposing same host port in one node.
+				if _, ok := pod.Spec.NodeSelector["nominated"]; ok {
+					pod.Spec.Containers[0].Ports = []v1.ContainerPort{{HostPort: 8080, ContainerPort: 8080}}
+				}
+				if _, err := cs.CoreV1().Pods(ns).Create(testCtx.Ctx, pod, metav1.CreateOptions{}); err != nil {
+					t.Fatalf("failed to create existing Pod %q: %v", pod.Name, err)
+				}
+
+				// Ensure the existing Pods are scheduled successfully except for the nominated pods.
+				if err := wait.Poll(1*time.Second, 20*time.Second, func() (bool, error) {
+					return podScheduled(cs, ns, pod.Name), nil
+				}); err != nil {
+					t.Logf("pod %q failed to be scheduled", pod.Name)
+				}
+
+				// Create Terminating Pods by deleting pods from cluster.
+				if pod.DeletionTimestamp != nil {
+					if err := cs.CoreV1().Pods(ns).Delete(context.TODO(), pod.Name, metav1.DeleteOptions{}); err != nil {
+						t.Fatalf("failed to delete existing Pod %q: %v", pod.Name, err)
+					}
+				}
+			}
+
+			// Create Pod to be scheduled.
+			if _, err := cs.CoreV1().Pods(ns).Create(testCtx.Ctx, tt.pod, metav1.CreateOptions{}); err != nil {
+				t.Fatalf("failed to create Pod %q: %v", tt.pod.Name, err)
+			}
+
+			// Ensure the Pod is scheduled successfully.
+			if err := wait.Poll(1*time.Second, 60*time.Second, func() (bool, error) {
+				return podScheduled(cs, ns, tt.pod.Name), nil
+			}); err != nil {
+				t.Errorf("pod %q failed to be scheduled: %v", tt.pod.Name, err)
+			}
+
+			// Lastly, verify pod gets scheduled to the expected node.
+			pod, _ := cs.CoreV1().Pods(ns).Get(context.TODO(), tt.pod.Name, metav1.GetOptions{})
```

Ah I forgot to comment on this: don't swallow err :)